### PR TITLE
[mariadb-connection-c] Version 3.3.0 requires zstd

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **mariadb-connection-c/3.3.3**

Since the version 3.0.0-rc, the project Maria DB Connector C needs zstd: https://mariadb.com/kb/en/mariadb-connector-c-330-release-notes/

It's not an optional requirement.

This PR is required by #20770

/cc @SpaceIm 

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
The way they found zstd in upstream CMakeLists is not very robust:
https://github.com/mariadb-corporation/mariadb-connector-c/blob/v3.3.1/CMakeLists.txt#L222
https://github.com/mariadb-corporation/mariadb-connector-c/blob/v3.3.1/cmake/FindZStd.cmake.

I'm pretty sure it will fail with msvc, mariadb shared and zstd static.

**Comment by SpaceIm** [comment]
In https://c3i.jfrog.io/c3i/misc/logs/pr/20860/1-windows-visual_studio/mariadb-connector-c/3.3.3/1efe6675d8dce08834aefd0019e7ab8a82ce3e1d-build.txt

```
-- Could NOT find ZSTD (missing: ZSTD_LIBRARIES) 
-- Looking for include file linux/limits.h
-- Looking for include file linux/limits.h - not found
-- Looking for include file signal.h
-- Looking for include file signal.h - found
-- Looking for include file ucontext.h
-- Looking for include file ucontext.h - not found
-- Check size of uint
-- Check size of uint - failed
-- TLS library/version: Schannel 10.0.20348
-- SYSTEM_LIBS ws2_32;advapi32;kernel32;shlwapi;crypt32;secur32
-- SYSTEM_LIBS: ws2_32;advapi32;kernel32;shlwapi;crypt32;secur32;secur32;ZLIB::ZLIB
-- Dynamic column API support: ON
-- SYSTEM processor: AMD64
-- MariaDB Connector/c configuration:
-- Static PLUGINS mysql_native_password;auth_gssapi_client;mysql_old_password;zlib;pvio_socket
-- Dynamic PLUGINS dialog;client_ed25519;caching_sha2_password;sha256_password;mysql_clear_password;remote_io;pvio_npipe;pvio_shmem
-- CPack generation: ZIP
-- SSL support: SCHANNEL Libs: secur32
-- Zlib support: ON
-- ZStd support: FALSE
```

**Review by jwillikers — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by SpaceIm** [comment]
Here is an intrusive-less solution:
- set `cmake_file_name` property of zstd dependency in CMakeDeps to `ZSTD`
- replace https://github.com/mariadb-corporation/mariadb-connector-c/blob/v3.3.1/CMakeLists.txt#L222 by `find_package(ZSTD REQUIRED CONFIG)`

**Review by jwillikers — APPROVED** [review]

File: recipes/mariadb-connector-c/all/conanfile.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def generate(self):
         tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
         tc.generate()
         deps = CMakeDeps(self)
+        deps.set_property("zstd", "cmake_file_name", "ZSTD")
```
Comment (SpaceIm) [inline recipes/mariadb-connector-c/all/conanfile.py:99]:
```suggestion
        if Version(self.version) >= "3.3":
            deps.set_property("zstd", "cmake_file_name", "ZSTD")
```

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
> An unexpected error happened and has been reported. Help is on its way! 🏇

v2 pipeline needs to be triggered again

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]
This is one of the cases against automatic new version merges, thanks for fixing it :)

**Pull request merged**